NDTensors/src/lib/TypeParameterAccessors/src/typeparamaccessor.jl

```diff
@@ -0,0 +1,23 @@
+# Relies on Julia internals
+Base.@assume_effects :terminates_globally to_datatype(type::UnionAll) =
+  Base.unwrap_unionall(type)
+Base.@assume_effects :terminates_globally to_unionall(
+  type::DataType, unionall_reference::Type
+) = Base.rewrap_unionall(type, unionall_reference)
```

Comment:
I found in my testing that `:foldable` made more things type stable than `:terminates_globally`. Curious if you see the same thing.

Reply:
This code is removed. I tried using `:foldable` before removing it and it did not help my type unstable portion of code